src/core/thread/mle_router.hpp

```diff
@@ -317,6 +317,17 @@ class MleRouter: public Mle
     Child *GetChildren(uint8_t *aNumChildren);
 
     /**
+     * This method sets the max children allowed value for this Thread interface.
+     *
+     * @param[in]  aMaxChildren  The max children allowed value.
+     *
+     * @retval  kThreadErrorNone           Successfully set the max.
+     * @retval  kThreadError_InvalidArgs   If @p aMaxChildren is not in the range [1, kMaxChildren].
+     *
+     */
+    ThreadError SetMaxAllowedChildren(uint8_t aMaxChildren);
```

Comment:
This can only be called when thread is disabled. That check is made in openthread.cpp. Since it can only be changed when disabled, no one should have any current references to the other children.